src/cfnlint/rules/__init__.py

```diff
@@ -474,22 +474,30 @@ def __init__(
             columnnumberend, filename, rule, message=None, rulematch_obj=None):
         """Init"""
         self.linenumber = linenumber
+        """Starting line number of the region this match spans"""
         self.columnnumber = columnnumber
+        """Starting line number of the region this match spans"""
         self.linenumberend = linenumberend
+        """Ending line number of the region this match spans"""
         self.columnnumberend = columnnumberend
+        """Ending column number of the region this match spans"""
         self.filename = filename
+        """Name of the filename associated with this match, or None if there is no such file"""
         self.rule = rule
+        """The rule of this match"""
         self.message = message  # or rule.shortdesc
+        """The message of this match"""
         if rulematch_obj:
             for k, v in vars(rulematch_obj).items():
                 if not hasattr(self, k):
                     setattr(self, k, v)
 
     def __repr__(self):
         """Represent"""
-        formatstr = u'[{0}] ({1}) matched {2}:{3}'
+        file_str = self.filename + ':' if self.filename else ''
```

Comment:
Whats this do?  Didn't change any tests but trying to understand

Reply:
Since `filename` can now take the value `None`, this makes it so that in such cases `filename` is omitted, so you would just get:
`"[E1012: Check if Refs exist] (Ref ServiceName not found as a resource or parameter) matched 7"` 
Instead of:
`"[E1012: Check if Refs exist] (Ref ServiceName not found as a resource or parameter) matched None:7"`

Reply:
Got it.